vnext/ReactUWP/Views/DynamicValueProvider.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "DynamicAutomationProperties.h"
+#include "DynamicValueProvider.h"
+
+#include <UI.Xaml.Controls.h>
+
+// Needed for latest versions of C++/WinRT
+#if __has_include("DynamicValueProvider.g.cpp")
+#include "DynamicValueProvider.g.cpp"
+#endif
+
+namespace winrt::PROJECT_ROOT_NAMESPACE::implementation {
+
+DynamicValueProvider::DynamicValueProvider(xaml::Automation::Peers::FrameworkElementAutomationPeer peer)
+    : m_peer(peer) {}
+
+// IValueProvider
+winrt::hstring DynamicValueProvider::Value() {
+  return GetAccessibilityValue(winrt::PROJECT_ROOT_NAMESPACE::AccessibilityValue::Text);
+}
+
+// The following methods are not React Native props, but we need them to implement IValueProvider.
+bool DynamicValueProvider::IsReadOnly() {
+  return false;
+}
+
+void DynamicValueProvider::SetValue(winrt::hstring const &value) {
+  try {
+    if (auto const &owner = m_peer.Owner()) {
+      DynamicAutomationProperties::DispatchAccessibilityAction(owner, L"setValue");
+    }
+  } catch (...) {
+  }
+}
+
+// Private Methods
+winrt::hstring DynamicValueProvider::GetAccessibilityValue(
+    winrt::PROJECT_ROOT_NAMESPACE::AccessibilityValue accValue) const {
+  try {
+    if (auto const &owner = m_peer.Owner()) {
+      switch (accValue) {
+        case winrt::PROJECT_ROOT_NAMESPACE::AccessibilityValue::Text:
```

Comment:
Text overrides value if present, otherwise, you should return the value converted to the string. This is what RN wants https://reactnative.dev/docs/accessibility#accessibilityvalue-android-ios, and very nicely, it matches what we do for real Sliders in XAML. So you should check if `text` is present, and use that, otherwise use the `now` if present, otherwise return empty string.

Reply:
Also you don't really need the full switch statement here.

Reply:
@jonthysell Right, RN Text overrides the RN Now (xaml calls both Value), so perhaps I should do the check in FrameworkElementViewManager::UpdateProperties(), where I'd ignore min/max/now, if text is present?
Code in DynamicValueProvider only gets called after a check for Text in DynamicAutomationPeer. If I cast Now to retur it as Text, I'm basically returning the same value from two different providers, since the RangePattern will also return it, no?